src/test/kotlin/org/rust/ide/intentions/MatchToIfLetIntentionTest.kt

```diff
@@ -25,39 +25,51 @@ class MatchToIfLetIntentionTest : RsIntentionTestBase(MatchToIfLetIntention::cla
         }
     """)
 
-    fun `test unavailable all void arms`() = doUnavailableTest("""
-        enum MyOption {
-            Nothing,
-            Some(x),
-        }
+    fun `test unavailability empty match`() = doUnavailableTest("""
+        enum MyOption { Some(u32) }
 
         fn main() {
-            let a = MyOption::Some(52);
+            let color = MyOption::Some(52);
 
-            /*caret*/match a {
-                MyOption::Some(x) => {}
-                Nothing => {}
-            }
+            /*caret*/match color {};
         }
     """)
 
-    fun `test unavailable all not void arms`() = doUnavailableTest("""
-        enum MyOption {
-            Nothing,
-            Some(x),
+    fun `test block expr without line`() = doAvailableTest("""
+        enum OptionColor {
+            NoColor,
+            Color(i32, i32, i32),
         }
 
         fn main() {
-            let a = MyOption::Some(52);
+            let color = OptionColor::Color(255, 255, 255);
 
-            match a {
-                MyOption::Some(x) => {42}
-                Nothing => {43}/*caret*/
-            }
+            /*caret*/match color {
+                OptionColor::Color(255, 255, 255) => { 1 },
+                OptionColor::Color(_, _, _) => { 2 }
+                OptionColor::NoColor => { 3 }
+            };
+        }
+    """, """
+        enum OptionColor {
+            NoColor,
+            Color(i32, i32, i32),
+        }
+
+        fn main() {
+            let color = OptionColor::Color(255, 255, 255);
+
+            if let OptionColor::Color(255, 255, 255) = color {
+                1
+            } else if let OptionColor::Color(_, _, _) = color {
+                2
+            } else if let OptionColor::NoColor = color {
+                3
+            };
         }
```

Comment:
Hmm, I wonder what should we do in this case. Introduce an artificial `else` branch? I don't think that we can just use `else` for the last branch, since the `let` pattern might introduce some variables used in the corresponding match arm.